app/controllers/orders/pending.js

```diff
@@ -39,14 +39,12 @@ export default Controller.extend({
 
   actions: {
     aliPayCheckout(order_identifier) {
-      this.loader.post(`/create_source/${order_identifier}`)
-        .then(charge => {
-          if (charge.status) {
-            this.notify.success('Payment has succeeded');
-            this.transitionToRoute('orders.view', order_identifier);
-          } else {
-            this.notify.error('Payment has failed');
-          }
+      return this.loader.load(`alipay/create_source/${order_identifier}`)
+        .then(res => {
+          window.location.replace(res.link);
+        })
+        .catch(() => {
+          this.notify.error(this.l10n.t('Could not communicate with the server'));
```

Comment:
Print the actual exception to the console too. If the server is returning an error message that can be parsed as json, parse it and display that instead of just saying `Could not communicate with the server`

Reply:
If there's any error from the server side, 
```
{
    "status": False,
    "error": {message}
}
```
Will output this.